code/frontend/src/pages/chat/Chat.tsx

```diff
@@ -155,41 +147,47 @@ const Chat = () => {
     fetchServerConfig();
   }, []);
 
-  const startSpeechRecognition = (
-    subscriptionKey: string,
-    serviceRegion: string
-  ) => {
+  const fetchSpeechToken = async (): Promise<{ token: string, region: string; }> => {
+    try {
+      const response = await fetch("/api/speech/token");
+      if (!response.ok) {
+        throw new Error("Network response was not ok");
```

Comment:
Should we log out the response to aid future debugging?

Reply:
Added